examples/opensora_hpcai/opensora/datasets/video_dataset.py

```diff
@@ -0,0 +1,212 @@
+import csv
+import logging
+import os
+import time
+
+import albumentations
+import cv2
+import numpy as np
+from decord import VideoReader
+
+import mindspore as ms
+
+logger = logging.getLogger()
+
+
+def create_video_transforms(h, w, interpolation="bicubic"):
+    """
+    h, w : target resize height, weight
+    """
+    # expect rgb image in range 0-255, shape (h w c)
+    from albumentations import CenterCrop, SmallestMaxSize
+
+    mapping = {"bilinear": cv2.INTER_LINEAR, "bicubic": cv2.INTER_CUBIC}
+    pixel_transforms = albumentations.Compose(
+        [
+            SmallestMaxSize(max_size=h, interpolation=mapping[interpolation]),
+            CenterCrop(h, w),
+        ],
+    )
+
+    return pixel_transforms
+
+
+class VideoDataset:
```

Comment:
For efficient data sampling, data fetching and preprocessing should be separated into different pipelines, as one is IO intensive, and the other is CPU intensive. I integrated such pipelines in [OpenSora v1.1 training](https://github.com/mindspore-lab/mindone/pull/494/files#diff-4af26d61026bd16cbca7135edf4e1fa4d75db7d4a8db947aed46d2c27c653710). Maybe we follow this deisgn?

Reply:
got it. It should be more efficient.  How is performance gain when using `dataloader.map` to do transformation instead? Is frame consistency preserved for random augmentation ?

Reply:
Unfortunately, frame consistency for random augmentations is not available in MindData yet. But it can be easily fixed by introducing custom transforms, i.e.:
```python
class HorizontalFlip:
    def __init__(self, p: float = 0.5):
        self._p = p

    def __call__(self, frames: np.ndarray) -> np.ndarray:
        if random.random() < self._p:
            return np.flip(frames, axis=-1)     # assuming HWC order
```

On the performance gain: didn't have time to measure, but theoretically, there should be a noticeable speed boost as videos can be read and processed in parallel rather than in sequence as it is done now (each worker must load a video first).